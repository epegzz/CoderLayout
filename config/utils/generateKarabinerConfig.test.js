const generateKarabinerConfig = require('./generateKarabinerConfig')

describe('generateKarabinerConfig', () => {
  test('works', () => {
    const karabinerLayers = {
      layer0: {
        // 'keypad_plus' or `keypad_minus`
        triggerKeys: ['keypad_plus', 'keypad_minus'],
        mapping: {
          a: 'left_control',
        }
      },
      layer1: {
        // 'keypad_plus' and `keypad_minus`
        triggerKeys: [['keypad_plus', 'keypad_minus']],
        mapping: {
          x: '1',
          y: ['2', { output: '2' }],
          z: { output: '3', modifiers: ['left_command'] }
        }
      },
      layer2: {
        triggerVariables: ['trigger_spacebar'],
        mapping: {
          a: 'left_control',
        }
      }
    }

    const keylayoutConfig = [
      { output: '1', virtualKeyName: 'n' },
      { output: '2', virtualKeyName: 'n', shift: true },
      { output: '3', virtualKeyName: 'm' },
    ]

    expect(generateKarabinerConfig({
      karabinerLayers,
      keylayoutConfig
    })).toMatchSnapshot()
  })
})
