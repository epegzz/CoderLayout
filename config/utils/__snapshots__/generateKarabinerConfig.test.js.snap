// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateKarabinerConfig works 1`] = `
Object {
  "layers": Object {
    "layer0": Object {
      "mapping": Array [
        Object {
          "from": "a",
          "to": Object {
            "keyName": "left_control",
            "modifiers": Array [],
          },
        },
      ],
      "triggerKeys": Array [
        "keypad_plus",
        "keypad_minus",
      ],
    },
    "layer1": Object {
      "mapping": Array [
        Object {
          "from": "x",
          "to": Object {
            "keyName": "n",
            "modifiers": Array [],
            "output": "1",
          },
        },
        Object {
          "from": "y",
          "to": Object {
            "keyName": "n",
            "modifiers": Array [
              "shift",
            ],
            "output": "2",
          },
        },
        Object {
          "from": "z",
          "to": Object {
            "keyName": "m",
            "modifiers": Array [
              "left_command",
            ],
            "output": "3",
          },
        },
      ],
      "triggerKeys": Array [
        Array [
          "keypad_plus",
          "keypad_minus",
        ],
      ],
    },
    "layer2": Object {
      "mapping": Array [
        Object {
          "from": "a",
          "to": Object {
            "keyName": "left_control",
            "modifiers": Array [],
          },
        },
      ],
      "triggerKeys": undefined,
    },
  },
  "triggerKeys": Array [
    "keypad_plus",
    "keypad_minus",
  ],
  "triggerVariables": Array [
    "trigger_spacebar",
  ],
}
`;
